azure-iot-device/azure/iot/device/common/http_transport.py

```diff
@@ -13,6 +14,10 @@
 logger = logging.getLogger(__name__)
 
 
+# NOTE: There should probably be a more global timeout configuration, but for now this will do.
```

Comment:
10 seconds seems a bit short.

Reply:
This is the same as a timeout for MQTT Subscribe/Unsubscribe and more than an MQTT Publish (which Paho times out after 5s).

I do agree that there should be a more uniform policy, but as the comment indicates, that would need to be a different (and larger) PR.